### ISSUE TITLE
[Transform] Support range aggregations

### ISSUE THREAD

**Issue by simianhacker** [elastic/elasticsearch#86367]
### Description

I'm currently working on transforming APM data into summary documents for implementing SLO's. The current latency for an APM transaction is stored using a `histogram` field. We would like to be able to create ratio for `good transactions / total transactions`. In the example below, a good transaction is anything under `200ms` which can be extracted from the histogram field using a `range` aggregation like `{ "range": { "field": "transaction.duration.histogram", "ranges": [{ "to": 200000 }] }` along with the total transactions like `{ "range": { "field": "transaction.duration.histogram", "ranges": [{ "from": 0 }] }`. These values are then stored as a `numerator` and a `denominator` using a `bucket_script`. Here is an example of the transform definition we would like to create.

```JSON
PUT _transform/slo_latency
{
  "source": {
    "index": "remote_cluster:metrics-apm*",
    "runtime_mappings": {
      "sloId": {
        "type": "keyword",
        "script": {
          "source": "emit('uuid-of-slo-definition')"
        }
      }
    }
  },
  "frequency": "1m",
  "dest": {
    "index": "slo_latency"
  },
  "sync": {
    "time": {
      "field": "@timestamp",
      "delay": "60s"
    }
  },
  "retention_policy": {
    "time": {
      "field": "@timestamp",
      "max_age": "30d"
    }
  },
  "pivot": {
    "group_by": {
      "transaction.name": {
        "terms": {
          "field": "transaction.name"
        }
      },
      "service.name": {
        "terms": {
          "field": "service.name"
        }
      },
      "sloId": {
        "terms": {
          "field": "sloId"
        }
      },
      "@timestamp": {
        "date_histogram": {
          "field": "@timestamp",
          "calendar_interval": "1m"
        }
      }
    },
    "aggregations": {
      "_numerator": {
        "range": {
          "field": "transaction.duration.histogram",
          "ranges": [
            {
              "to": 200000
            }
          ]
        }
      },
      "_demoninator": {
        "range": {
          "field": "transaction.duration.histogram",
          "ranges": [
            {
              "from": 0
            }
          ]
        }
      },
      "numerator": {
        "bucket_script": {
          "buckets_path": {
            "numerator": "_numerator['*-200000.0']>_count"
          },
          "script": "params.numerator"
        }
      },
      "denominator": {
        "bucket_script": {
          "buckets_path": {
            "denominator": "_demoninator['0.0-*']>_count"
          },
          "script": "params.denominator"
        }
      }
    }
  }
}
```

**Comment by benwtrent** [comment]
@simianhacker 
A `filter` aggregation doesn't work in this case?

Adding range aggregation support is indeed possible.

**Comment by simianhacker** [comment]
I will play with the filter aggregation and report back.

**Comment by simianhacker** [comment]
@benwtrent Histograms fields are not supported by range filters, only range aggregations.

**Comment by benwtrent** [comment]
OK, lets see about supporting a range aggregation in transforms :)

**Comment by przemekwitek** [comment]
> OK, lets see about supporting a range aggregation in transforms :)

I'm currently testing how easy would it be to add `range` to transform-supported aggs.

**Comment by przemekwitek** [comment]
FTR, support for `range` aggregation in transform is now merged in `8.3`.

@simianhacker, feel free to try it out.
